sklearn/tree/_criterion.pyx

```diff
@@ -784,7 +769,7 @@ cdef class RegressionCriterion(Criterion):
                 w = sample_weight[i]
 
             for k in range(self.n_outputs):
-                y_ik = y[i * y_stride + k]
+                y_ik = self.y[i, k]
```

Comment:
The python overhead from this may be cause `self.y` is defined and called in `__cinit__`

In the cython docs a cdef is defined at the class level: https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#existing-pointers-instantiation

Something like this:

```python
cdef class Penguin:
    cdef int food

    def __cinit__(self, food):
        self.food = food

    def __init__(self, food):
        print("eating!")
```

Reply:
It actually is typed in criterion :/